provider/lxd/storage.go

```diff
@@ -188,21 +189,41 @@ func (e *lxdStorageProvider) FilesystemSource(cfg *storage.Config) (storage.File
 }
 
 func ensureLXDStoragePool(env *environ, cfg *lxdStorageConfig) error {
-	createErr := env.raw.CreateStoragePool(cfg.lxdPool, cfg.driver, cfg.attrs)
-	if createErr == nil {
+	err := env.raw.CreateStoragePool(cfg.lxdPool, cfg.driver, cfg.attrs)
+	if err == nil {
 		return nil
-	}
-	// There's no specific error to check for, so we just assume
-	// that the error is due to the pool already existing, and
-	// verify that. If it doesn't exist, return the original
-	// CreateStoragePool error.
+	} else if ok := strings.Contains(err.Error(), "pool already exists"); !ok {
+		return errors.Annotatef(err, "creating storage pool %q", cfg.lxdPool)
+	}
+
+	// If the pool already exists we must check to see if its configuration
+	// matches the pool we are attempting to create. We run several attempts
+	// to find the storage pool to guard against race conditions that can be
+	// encountered when multiple pools are created concurrently.
+	logger.Infof("Attempted to create storage pool %q but it already exists", cfg.lxdPool)
+	var pool api.StoragePool
+	retryCount := 5
+	for retryCount > 0 {
+		pool, err = env.raw.StoragePool(cfg.lxdPool)
+		if errors.IsNotFound(err) {
```

Comment:
There are already two libraries that we use for controlled retry.

For example, github.com/juju/retry in provider/azure/environ.go line 918
And gopkg.in/retry.v1 in api/apiclient.go.